mountinfo/openat2_linux.go

```diff
@@ -0,0 +1,44 @@
+package mountinfo
+
+import (
+	"unsafe"
+
+	"golang.org/x/sys/unix"
+)
+
+type openHow struct {
+	Flags   uint64
+	Mode    uint64
+	Resolve uint64
+}
+
+const sizeofOpenHow = 0x18
+
+const (
+	RESOLVE_BENEATH       = 0x8
+	RESOLVE_IN_ROOT       = 0x10
+	RESOLVE_NO_MAGICLINKS = 0x2
+	RESOLVE_NO_SYMLINKS   = 0x4
+	RESOLVE_NO_XDEV       = 0x1
+)
+
+// FIXME: sycall numbers are arch-specific, this won't work on e.g. MIPS
+const SYS_OPENAT2 = 437
```

Comment:
Actually, `openat2` should have the same syscall number on all architectures (except maybe Alpha). There was a very recently push to make all newer syscalls have the same syscall numbers.

Reply:
This is temp POC code, I'll remove it now once https://go-review.googlesource.com/c/sys/+/253057/ is merged